src/server/game/Entities/Player/Player.cpp

```diff
@@ -23968,8 +23968,11 @@ void Player::SetViewpoint(WorldObject* target, bool apply)
         // farsight dynobj or puppet may be very far away
         UpdateVisibilityOf(target);
 
-        if (target->isType(TYPEMASK_UNIT) && target != GetVehicleBase())
+        if (target->isType(TYPEMASK_SEER) && target != GetVehicleBase())
```

Comment:
This change is not valid. TYPEMASK_SEER includes TYPEMASK_DYNAMICOBJECT but the code uses a static_cast<Unit*>. DynamicObject is not Unit and will crash.